src/extensions/context.ts

```diff
@@ -0,0 +1,107 @@
+import { CycliToolbox, PackageManager } from '../types'
+import { LOCK_FILE_TO_MANAGER } from '../constants'
+import { lookItUpSync } from 'look-it-up'
+import { basename, join, relative } from 'path'
+
+module.exports = (toolbox: CycliToolbox) => {
+  const { filesystem } = toolbox
+
+  // State for caching the context
+  let packageManager: PackageManager | undefined = undefined
+  let repoRoot: string | undefined = undefined
+  let packageRoot: string | undefined = undefined
+
+  const getPackageManager = (repoRoot: string): PackageManager => {
+    if (!packageManager) {
+      const lockFiles =
+        filesystem
+          .list(repoRoot)
+          ?.filter((fileName: string) =>
+            Object.keys(LOCK_FILE_TO_MANAGER).includes(fileName)
+          ) || []
+
+      if (lockFiles.length == 0) {
+        throw Error(
+          [
+            'No lock file found in repository root directory. Are you sure you are in a project directory?',
+            'Make sure you generated lock file by installing project dependencies.',
+          ].join('\n')
+        )
+      }
+
+      if (lockFiles.length > 1) {
+        toolbox.interactive.warning(
+          `Detected more than one lock file in repository root directory. Proceeding with ${lockFiles[0]}.`
+        )
+      }
+
+      packageManager = LOCK_FILE_TO_MANAGER[lockFiles[0]]
+    }
+
+    return packageManager as PackageManager
```

Comment:
I guess instead of casting you could add there a check if it's filled and if not throw an error + would be good to see what will happen with bun or pnpm - I mean add some test for that case

Reply:
I know it should throw earlier but just in case and to satisfy TS (you can add comment too)